src/main/java/com/ozonehis/eip/odooopenmrs/client/OdooClient.java

```diff
@@ -0,0 +1,133 @@
+package com.ozonehis.eip.odooopenmrs.client;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.emptyMap;
+import static java.util.Collections.singletonList;
+import static java.util.Collections.singletonMap;
+
+import com.ozonehis.eip.odooopenmrs.Constants;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.xmlrpc.XmlRpcException;
+import org.apache.xmlrpc.client.XmlRpcClient;
+import org.apache.xmlrpc.client.XmlRpcClientConfigImpl;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Getter
+@Component
+public class OdooClient {
+
+    @Value("${odoo.baseUrl}")
+    private String url;
+
+    @Value("${odoo.database}")
+    private String database;
+
+    @Value("${odoo.username}")
+    private String username;
+
+    @Value("${odoo.password}")
+    private String password;
+
+    private Integer uid;
+
+    private XmlRpcClient client;
+
+    private static final String SERVER_OBJECT_URL = "%s/xmlrpc/2/object";
+
+    private static final String SERVER_COMMON_URL = "%s/xmlrpc/2/common";
+
+    public OdooClient() {}
+
+    public void init() throws IOException {
+        client = new XmlRpcClient() {
+            {
+                setConfig(new XmlRpcClientConfigImpl() {
+
+                    {
+                        setEnabledForExtensions(true);
+                        setServerURL(new URL(String.format(SERVER_OBJECT_URL, getUrl())));
+                    }
+                });
+            }
+        };
+    }
+
+    public void authenticate() throws MalformedURLException {
+        // TODO: Where should we place the init logic
+        try {
+            init();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+        XmlRpcClientConfigImpl common_config = new XmlRpcClientConfigImpl();
+        common_config.setServerURL(new URL(String.format(SERVER_COMMON_URL, getUrl())));
+
+        try {
+            uid = (Integer) client.execute(
+                    common_config, "authenticate", asList(getDatabase(), getUsername(), getPassword(), emptyMap()));
+        } catch (XmlRpcException e) {
+            log.error("Cannot authenticate to Odoo server error: {}", e.getMessage());
+            throw new RuntimeException("Cannot authenticate to Odoo server");
+        }
+    }
+
+    public Integer create(String model, List<Map<String, Object>> dataParams)
```

Comment:
Is there a reason for favoring `Integer` over `int` here and boxed types over primitives in general?

Reply:
Tried to keep the client consistent with [Odoo documentation](https://www.odoo.com/documentation/17.0/developer/reference/external_api.html#create-records)